File: dotnet/src/Core/Models/Entities/Function.cs
using Agience.Core.Models.Entities.Abstract;
using System.Reflection.Metadata;
using System.Text.Json.Serialization;

namespace Agience.Core.Models.Entities
{
    public class Function : DescribedEntity
    {
        //[JsonPropertyName("connection_id")]
        //public string? ConnectionId { get; set; }

        [JsonPropertyName("instruction")]
        public string? Instruction { get; set; }

        [JsonPropertyName("inputs")]
        public virtual List<Parameter> Inputs { get; set; } = new();
        
        [JsonPropertyName("outputs")]
        public virtual List<Parameter> Outputs { get; set; } = new();

        //[JsonPropertyName("execution_settings")]
        //public Dictionary<string, PromptExecutionSettings> ExecutionSettings { get; set; } = [];
    }
}

File: python/src/core/models/entities/function.py
from pydantic import Field
from typing import Optional, List
from models.entities.abstract.described_entity import DescribedEntity
from models.entities.parameter import Parameter


class Function(DescribedEntity):
    instruction: Optional[str] = Field(None, alias="instruction")
    inputs: List[Parameter] = Field(default_factory=list, alias="inputs")
    outputs: List[Parameter] = Field(default_factory=list, alias="outputs")

    class Config:
        allow_population_by_field_name = True
